gnocchi-core/src/test/scala/net/fnothaft/gnocchi/association/LogisticValidationRegressionSuite.scala

```diff
@@ -68,5 +68,52 @@ class LogisticValidationRegressionSuite extends GnocchiFunSuite {
     }).count(b => { b }))
     assert(predictionResult.sameElements(expectedResults))
   }
+
+  sparkTest("Test logistic regression prediction on binary data with a different threshold") {
+    // Reads in the binaryClean data, which is the same as the above data but removes any
+    // duplicate samples. The Fourth row (0 indexed) is the prediction results generated
+    // from a logisitic regression model in excel with a threshold classification value of 0.3
+    val pathToFile = ClassLoader.getSystemClassLoader.getResource("binaryThreshold3.csv").getFile
+    val csv = sc.textFile(pathToFile)
+    val data = csv.map(line => line.split(",").map(elem => elem.toDouble)) //get rows
+
+    // transform it into the right format
+    val sampleObservations = data.map(row => {
+      val geno: Double = row(0)
+      val covars: Array[Double] = row.slice(1, 3)
+      val phenos: Array[Double] = Array(row(3)) ++ covars
+      (geno, phenos, "")
+    }).collect()
+
+    // generate array of expected results for each sample, based on given phenotype
+    val expectedResults = data.map(row => {
+      val actual: Double = row(3)
+      val predicted: Double = row(4)
+      ("", (actual, predicted))
+    }).collect()
+
+    val fakeVariant = new Variant()
+    val fakePhenotype = "acceptance"
+    // Our known fitting model
+
+    val weights = Array(-3.351609998, 0.002338182, 0.742468226, -0.568764356)
+    val statistics = Map("weights" -> weights)
+    val assoc = Association(fakeVariant, "", 0.0, statistics)
+
+    // Array[(String, Double)] where String is sampleid and Double is predicted value
+    val predictionResult = AdditiveLogisticEvaluation.predictSite(sampleObservations, assoc, 0.3)
+
+    // Assert that the predictions result in the same as the actual phenotype (on dummy set)
+    for (i <- predictionResult.indices) {
+      if (predictionResult(i) != expectedResults(i)) {
+        print("Error --> ")
```

Comment:
these debugging statements shouldn't be left in, make assertions and let the test framework report errors